spock-core/src/main/java/org/spockframework/util/ThreadSupport.java

```diff
@@ -0,0 +1,26 @@
+package org.spockframework.util;
+
+import java.lang.reflect.Method;
+
+public class ThreadSupport {
+  private static final @Nullable Class<?> THREAD_OF_VIRTUAL_CLASS = ReflectionUtil.loadClassIfAvailable("java.lang.Thread$Builder$OfVirtual");
+  private static final @Nullable Method OF_VIRTUAL = THREAD_OF_VIRTUAL_CLASS != null ? ReflectionUtil.getMethodByName(Thread.class, "ofVirtual") : null;
```

Comment:
Is the Class `OfVirtual` part of the API, I would rather check for the method `Thread.ofVirtual()`, because if the `OfVirtual` class is not part of the API, it could change in future versions, rendering the check here invalid.

Reply:
I'm pretty sure it is part of the API, and in any case, it would be `null`. Either way, the null-check is not strictly necessary.